src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -1124,20 +1124,97 @@ protected function xmlToProps($xml)
     protected function xmlToColumns($xml, $propertyNames)
     {
         $data = new stdClass();
-        $dom = new DOMDocument('1.0');
-        $dom->loadXML($xml);
-        $dom->formatOutput = true;
-        $xpath = new DOMXPath($dom);
 
-        foreach ($propertyNames as $propertyName) {
-            $els = $xpath->query(sprintf('.//sv:property[@sv:name="%s"]', $propertyName));
-            if ($els->length === 0) {
-                continue;
+        $lastPropertyName = null;
+        $lastPropertyType = null;
+        $lastPropertyMultiValued = null;
+        $currentTag = null;
+        $currentValues = [];
+        $currentValue = null;
+
+        $parser = xml_parser_create();
+        xml_set_element_handler(
+            $parser,
+            function ($parser, $name, $attrs) use ($propertyNames, &$lastPropertyName, &$lastPropertyType, &$lastPropertyMultiValued, &$currentTag) {
+                $currentTag = $name;
+
+                if ($name === 'SV:PROPERTY') {
```

Comment:
in the interest of early return, we could say `if ('SV:PROPERTY' !== $name) { return`